### PR TITLE
Tracing: Document Tracer and Span interfaces

### PR DESCRIPTION
**Pull request by sakjur**
**What this PR does / why we need it**:

Adds documentation for the current way the Tracer and Span interfaces work.

**Special notes for your reviewer**:

I talked a bit with @idafurjes and @zserge about this, I think we'll want to modify some of these methods. I believe there's value in documenting the current behavior without trying to simultaneously improving things, especially considering the discrepancies between OTel and OpenTracing 🤔

### CODE REVIEW

**Review by mildwonkey — APPROVED** [review]
Enthusiastically agree - documentation now is better than later when the methods change.

File: pkg/infra/tracing/tracing.go [hunk 1]
```diff
@@ -20,13 +17,67 @@ import (
...
+	"github.com/grafana/grafana/pkg/setting"
 )
 
 const (
 	envJaegerAgentHost = "JAEGER_AGENT_HOST"
 	envJaegerAgentPort = "JAEGER_AGENT_PORT"
 )
 
+// Tracer defines the service used to create new spans.
+type Tracer interface {
```
Comment (idafurjes) [inline pkg/infra/tracing/tracing.go:32]: maybe we should rename this file to opentracing_tracing.go, I would leave the Tracer in the opentelemetry_tracer.go file, maybe would consider moving other stuff rather there

Comment (sakjur) [reply pkg/infra/tracing/tracing.go:32]: @idafurjes We should probably split out all the OpenTracing specifics into one such file, these two interfaces are for both and provided by `ProvideService(…) Tracer` in the _tracing.go_-file, so I found it confusing to have them defined in the OTel file? 🤔

Comment (idafurjes) [reply pkg/infra/tracing/tracing.go:32]: or we deprecate opentracing

Comment (sakjur) [reply pkg/infra/tracing/tracing.go:32]: 🙂

Could we use OTel in the background for OpenTracing?

Comment (idafurjes) [reply pkg/infra/tracing/tracing.go:32]: even if we could, it would be painful

Comment (sakjur) [reply pkg/infra/tracing/tracing.go:32]: :+1: Ok, let's not do that then

**Pull request merged**